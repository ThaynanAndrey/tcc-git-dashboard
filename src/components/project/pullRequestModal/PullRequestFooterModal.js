import React from 'react';

/**
 * Stateless component with the function to presenting the Footer Modal.
 * 
 * @author Thaynan Nunes
 */
const PullRequestFooterModal = ({ closePullRequestDetails }) => {
    return (
        <div className="modal-footer">
            <div className="modal-close waves-effect waves-green btn-flat"
                onClick={() => closePullRequestDetails()}>
                Fechar
            </div>
        </div>
    );
}

export default PullRequestFooterModal;