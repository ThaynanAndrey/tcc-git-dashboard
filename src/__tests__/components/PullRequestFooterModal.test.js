import React from 'react';
import { mount, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import PullRequestFooterModal from '../../components/project/pullRequestModal/PullRequestFooterModal';

describe('Test PullRequestFooterModal', () => {
  let tree;
  let closePullRequestDetails;

  beforeEach(() => {
    closePullRequestDetails = jest.fn();  

    tree = mount(
        <PullRequestFooterModal
        closePullRequestDetails={closePullRequestDetails}
        />
    );
  });

  it('renders correctly a component to show PullRequestFooterModal', () => {
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders correctly a component to show PullRequestFooterModal with correct props', () => {
    expect(tree.prop('closePullRequestDetails')).toEqual(closePullRequestDetails);
  });

  it('renders correctly a component to show PullRequestFooterModal with correct classes', () => {
    expect(tree.find('.modal-footer').exists()).toBeTruthy();
    expect(tree.find('.modal-close.waves-effect.waves-green.btn-flat').exists()).toBeTruthy();
  });
});