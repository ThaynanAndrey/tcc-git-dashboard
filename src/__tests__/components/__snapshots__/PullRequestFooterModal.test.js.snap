// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PullRequestFooterModal renders correctly a component to show PullRequestFooterModal 1`] = `
<PullRequestFooterModal
  closePullRequestDetails={[MockFunction]}
>
  <div
    className="modal-footer"
  >
    <div
      className="modal-close waves-effect waves-green btn-flat"
      onClick={[Function]}
    >
      Fechar
    </div>
  </div>
</PullRequestFooterModal>
`;
